nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/authorization/AuthorizeControllerServiceReference.java

```diff
@@ -57,8 +58,12 @@ public static void authorizeControllerServiceReferences(final Map<String, String
                     if (!Objects.equals(currentValue, proposedValue)) {
                         // ensure access to the old service
                         if (currentValue != null) {
-                            final Authorizable currentServiceAuthorizable = lookup.getControllerService(currentValue).getAuthorizable();
-                            currentServiceAuthorizable.authorize(authorizer, RequestAction.READ, user);
+                            try {
+                                final Authorizable currentServiceAuthorizable = lookup.getControllerService(currentValue).getAuthorizable();
+                                currentServiceAuthorizable.authorize(authorizer, RequestAction.READ, user);
+                            } catch (ResourceNotFoundException e) {
+                                // ignore if the resource is not found, if currentValue was previously deleted, it should not stop assignment of proposedValue
```

Comment:
I wonder if instead of raising and then catching exception and do nothing we should instead introduce  ```canLocate(id)``` or ```isLocatable```?